docs/connectors/index.mdx

```diff
@@ -1,11 +1,170 @@
-# Data Sources and Sinks
+# Connectors: connect to data sources and sinks
 
 A Feldera pipeline can process data from multiple heterogeneous sources and
-produce outputs to multiple heterogeneous destinations.  To this end it relies
-on a growing library of input and output connectors.
+produce outputs to multiple heterogeneous destinations. To this end it relies
+on a growing library of **[input](sources)** and **[output](sinks)** connectors.
 
-```mdx-code-block
-import DocCardList from '@theme/DocCardList';
+:::caution
 
-<DocCardList />
+The connectors API is still evolving and might see backwards incompatible changes.
+
+:::
+
+## Basics
+
+The connector object is the abstraction used to define data coming in from data sources
+and going out to data sinks.
+
+* The program of a pipeline defines the tables and views it has,
+  to which input and output connectors can respectively be attached.
+* A connector must explicitly be attached to a table (for input)
+  or view (for output) of a pipeline.
+* A connector can be attached to the tables/views of one or more pipelines,
+  and can even be attached to multiple tables (for input) or views (for output)
+  within the same pipeline.
+* Connectors are managed through the API under the **/v0/connectors/...** endpoints.
+  They can be retrieved
+  [individually](https://www.feldera.com/api/fetch-a-connector-by-name) or
+  [as a list](https://www.feldera.com/api/fetch-connectors-optionally-filtered-by-name-or-id),
+  [created](https://www.feldera.com/api/create-a-new-connector),
+  [fully modified](https://www.feldera.com/api/create-or-replace-a-connector),
+  [partially modified](https://www.feldera.com/api/update-the-name-description-and-or-configuration-of-a-connector) and
+  [deleted](https://www.feldera.com/api/delete-an-existing-connector).
+* If changes are made to the connectors of a running pipeline, they will only be
+  applied when the pipeline is restarted.
+
+> **Note:** these basics apply to all connectors **except** the
```

Comment:
I am not sure we even need to expose the fact that HTTP connectors exist, since they are only used via the `ingress`/`egress` API.

Reply:
I think it's better to present them as a connector that's always enabled for convenience and requires no user-configuration.

Reply:
I agree, I think we also might want to consider architecturally enforce it as such, which would fit well with both informing the user as well as having it included seamlessly into table/view statistics. Currently they are ephemeral if I understand correctly though? There is some performance trade-off there of simultaneous HTTP requests vs. a single large one streaming, and as well how to forward to the pipeline.

Reply:
Problem is that every HTTP connection requires a new connector instance.  But as far as docs go, I see no value for the reader in knowing that there is a connector handling HTTP ingress/egress requests, since that doesn't manifest externally in any way.  This is not a big deal, I will leave it up to @snkas to decide how he wants this.

Reply:
I will leave it in connector docs for now, as it is in general a concept we need to explain as it is used widely -- the connectors documentation seems to be the most appropriate place for now.